client/components/SignUpForm/SignUpDumb.js

```diff
@@ -0,0 +1,174 @@
+import React, { Component } from 'react';
+import { Button, Form, Grid, Header, Message } from 'semantic-ui-react';
+import { Link } from 'react-router-dom';
+import Axios from 'axios';
+
+class SignUpForm extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      values: {
+        firstName: '',
+        lastName: '',
+        imageURL: '',
+        zipcode: '',
+        username: '',
+        email: '',
+        password: '',
+      },
+      errors: {},
+    };
+  }
+
+  handleChange = e => {
+    const { name, value } = e.target;
+
+    this.setState(state => ({
+      ...state,
+      values: { ...state.values, [name]: value },
+    }));
+  };
+
+  handleSubmit = async e => {
+    e.preventDefault();
+
+    try {
+      const { data } = await Axios.post(
+        '/auth/local/signup',
+        this.state.values,
+        {
+          validateStatus: function(status) {
+            return status === 200 || status === 401;
+          },
+        }
+      );
+
+      if (data.error) {
+        this.setState(state => ({
+          ...state,
+          errors: { ...state.errors, ...data.error },
+        }));
+      } else if (data.errors) {
+        this.setState(state => ({
+          ...state,
+          errors: { ...state.errors, ...data.errors },
+        }));
+      } else {
+        this.props.gotUser(data);
+      }
+    } catch (error) {
+      console.error(error);
+    }
+  };
+
+  render() {
+    const { values, errors } = this.state;
+
+    return (
+      <Grid
+        textAlign="center"
+        style={{ height: '85vh' }}
+        verticalAlign="middle"
+      >
+        <Grid.Column style={{ maxWidth: 450 }}>
+          <Header as="h2" color="teal" textAlign="center">
+            Sign Up
+          </Header>
+          <Form size="large" onSubmit={this.handleSubmit}>
+            <Form.Input
+              fluid
+              icon="user"
```

Comment:
What does fluid do?

Reply:
I think it has to do with being responsive? not super sure. It was in the demo code.

Reply:
i believe it just means it takes up the width of its container